frame/utility/src/lib.rs

```diff
@@ -113,8 +113,12 @@ pub mod pallet {
 		BatchInterrupted { index: u32, error: DispatchError },
 		/// Batch of dispatches completed fully with no error.
 		BatchCompleted,
+		/// Batch of dispatches complete but has errors. Index of failing dispatches.
+		BatchCompletedWithErrors { indexes: Vec<u32> },
```

Comment:
Why is this Event needed? The `ItemFailed` should already give you that info, right?

Reply:
In my understanding, `BatchCompleted` and `BatchInterrupted` is the result of a `batch` call itself

Consider `batch`, each item success would get `ItemCompleted`, when all item processed, it deposit `BatchCompleted` too.

As for `BatchCompletedWithErrors`, I'm following the same pattern of `BatchCompleted` and `BatchCompleted`, it's the result of the `force_batch`

Reply:
I think the `BatchCompleted` only contains an error index since there is no `ItemFailed` emitted by the `batch` call.  
Anyway, let's see what the others think.

Reply:
I dont think is it really bad or good. Seems maybe redundant since we emit an error already at each failing extrinsic, but this is maybe convenient. So, sorry, but my opinion is I don't really have a strong opinion.

Reply:
This is just redundant and just blows up the state for nothing. As we already emit an event per item, it doesn't make any sense to emit the same information again at the end.

Reply:
> This is just redundant and just blows up the state for nothing. As we already emit an event per item, it doesn't make any sense to emit the same information again at the end.

What about keep `BatchCompletedWithErrors` event but remove `indexes` field?

Reply:
yeah that sounds better

Reply:
Sure, that seems better.

Reply:
changed.